CONTRIBUTING.md

```diff
@@ -132,8 +131,39 @@ If it has someone assigned to it then that person is the person responsible for
 
 At this time we do not accept PRs for adding additional language translations as we have an automated process for generating our translations. This is because translations require constant care as new strings are added and changed in the code base. Because of this the .po files are overwritten during every translation release cycle. We also can't support a lot of translations on AWX as its an open source project and each language adds time and cost to maintain. If you would like to see AWX translated into a new language please create an issue and ask others you know to upvote the issue. Our translation team will review the needs of the community and see what they can do around supporting additional language.
 
-If you find an issue with an existing translation, please see the [Reporting Issues](#reporting-issues) section to open an issue and our translation team will work with you on a resolution. 
+If you find an issue with an existing translation, please see the [Reporting Issues](#reporting-issues) section to open an issue and our translation team will work with you on a resolution.
+
+### Common Git workflows
+
+In general there are two separate git workflows to follow when contributing code for a net new feature to AWX. One for keeping a feature branch up to date with devel, and another, used by individual contributors to keep their working branch up to date with the feature branch they are working on. When initiating work on a net new feature a new feature branch should be checked out from awx/devel and branch protection should be activated. Checking out the new branch, and activating branch protection must be done by someone with admin privileges on the AWX repo. This branch will be used by the developers working on the feature and merging work into the feature branch will allow the developers to work quickly without impacting devel. This also means that those working on the feature can decide what requirements need to pass before merging work into the feature branch. Merge meeting, QE sign off etc. can all be done at 1 time when the feature is ready to merge into awx/devel.
+
+It is important that the feature branch be kept up-to-date with awx/devel to limit merge conflicts that might arise when merging the feature into devel. To do this someone with admin privileges on the AWX repo needs to remove branch protection and then follow these steps (In this example I will use origin as the remote name for AWX):
```

Comment:
I'd recommend using `upstream` instead of `origin`.  I think the most typical setup for dev checkouts on our team has origin == the developer's fork and upstream == main fork.